cou/cli.py

```diff
@@ -42,110 +71,333 @@ def parse_args(args: Any) -> argparse.Namespace:
+        if parsed_args.command == "help":
+            match parsed_args.subcommand:
+                case "run":
+                    run_parser.print_help()
+                case "plan":
+                    plan_parser.print_help()
+                case "all":
+                    parser.print_help()
+            sys.exit(0)
+
+        return parsed_args
+    except argparse.ArgumentError as exc:
+        print(f"Error parsing arguments: {exc}\n")
+        print("See 'cou help' for more information.")
+        sys.exit(1)
 
-    :returns: Nothing: This function is executed for its side effect
-    :rtype: None
+
+def get_log_level(quiet: bool = False, verbosity: int = 0) -> str:
+    """Get a log level based on input options.
+
+    :param quiet: Whether to run COU in quiet mode.
+    :type quiet: bool
+    :param verbosity: Verbosity level based on user's input.
+    :type verbosity: int
+    :return: Log level.
+    :rtype: str
     """
-    log_formatter_file = logging.Formatter(
-        fmt="%(asctime)s [%(name)s] [%(levelname)s] %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
-    )
-    log_formatter_console = logging.Formatter(
-        fmt="%(asctime)s [%(levelname)s] %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
-    )
-    root_logger = logging.getLogger()
-    root_logger.setLevel("DEBUG")
+    if quiet:
+        return "CRITICAL"
+    return VERBOSITY_LEVEL[verbosity] if verbosity <= 4 else VERBOSITY_LEVEL[4]
+
+
+async def analyze_and_plan(model_name: Optional[str] = None) -> UpgradeStep:
+    """Analyze cloud and generate the upgrade plan with steps.
+
+    :param model_name: Model name inputted by user.
+    :type model_name: Optional[str]
+    :return: Generated upgrade plan.
+    :rtype: UpgradeStep
+    """
+    model_name = model_name or await juju_utils.get_current_model_name()
+    logger.info("Using model: %s", model_name)
+
+    progress_indicator.start("Analyzing cloud...")
+    analysis_result = await Analysis.create(model_name)
+    progress_indicator.succeed()
+    logger.info(analysis_result)
 
-    # handler for the log file. Log level is DEBUG
-    time_stamp = datetime.now().strftime("%Y%m%d%H%M%S")
-    file_name = f"{COU_DIR_LOG}/cou-{time_stamp}.log"
-    pathlib.Path(COU_DIR_LOG).mkdir(parents=True, exist_ok=True)
-    log_file_handler = logging.FileHandler(file_name)
-    log_file_handler.setFormatter(log_formatter_file)
+    progress_indicator.start("Generating upgrade plan...")
+    upgrade_plan = await generate_plan(analysis_result)
+    progress_indicator.succeed()
+
+    return upgrade_plan
+
+
+async def get_upgrade_plan(model_name: Optional[str] = None) -> None:
+    """Get upgrade plan and print to console.
+
+    :param model_name: Model name inputted by user.
+    :type model_name: Optional[str]
+    """
+    upgrade_plan = await analyze_and_plan(model_name)
+    logger.info(upgrade_plan)
+    print(upgrade_plan)  # print plan to console even in quiet mode
```

Comment:
So if I run in in -vvv, I will see the plan twice?

Reply:
I think it should be debug or trace.

Reply:
I personally think this can stay at info level, similar as the analyze result, because they provide cloud-related information rather than program implementation details.